br/pkg/task/restore.go

```diff
@@ -633,18 +642,67 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		}
 	}
 
+	if client.IsIncremental() {
+		// don't support checkpoint for the ddl restore
+		cfg.UseCheckpoint = false
+	}
+	tree, gcID, err := client.InitCheckpoint(ctx, s, &restoreTS, cfg.UseCheckpoint)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	var gcTTL int64 = utils.DefaultBRGCSafePointTTL
+	if cfg.UseCheckpoint {
+		gcTTL = utils.DefaultCheckpointGCSafePointTTL
+	}
 	sp := utils.BRServiceSafePoint{
 		BackupTS: restoreTS,
-		TTL:      utils.DefaultBRGCSafePointTTL,
-		ID:       utils.MakeSafePointID(),
+		TTL:      gcTTL,
+		ID:       gcID,
 	}
 	g.Record("BackupTS", backupMeta.EndVersion)
 	g.Record("RestoreTS", restoreTS)
 
+	restoreSchedulers, err := restorePreWork(ctx, client, mgr, true)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	cctx, gcSafePointKeeperCancel := context.WithCancel(ctx)
+	gcSafePointKeeperRemovable := false
+	defer func() {
+		if cfg.UseCheckpoint {
+			// need to flush the whole checkpoint data so that br can quickly jump to
+			// the log kv restore step when the next retry.
+			if len(cfg.FullBackupStorage) > 0 || !gcSafePointKeeperRemovable {
+				log.Info("wait for flush checkpoint...")
+				client.WaitForFinishCheckpoint(ctx)
+			}
+			// don't reset the gc-safe-point and pd scheduler if checkpoint mode is used and restored is not finished
+			if !gcSafePointKeeperRemovable {
+				log.Info("skip removing gc-safepoint keeper and pd schehduler for next retry", zap.String("gc-id", sp.ID))
```

Comment:
the gc safepoint is only to avoid checksum taking too much time, so no need to keep gc safepoint？